include/cuco/detail/probing_scheme_impl.inl

```diff
@@ -120,8 +120,9 @@ __host__ __device__ constexpr auto linear_probing<CGSize, Hash>::operator()(
   Extent upper_bound) const noexcept
 {
   using size_type = typename Extent::value_type;
+  using cg_type   = cooperative_groups::thread_block_tile<cg_size>;
   return detail::probing_iterator<Extent>{
-    cuco::detail::sanitize_hash<size_type>(hash_(probe_key) + g.thread_rank()) % upper_bound,
+    cuco::detail::sanitize_hash<cg_type, size_type>(g, hash_(probe_key)) % upper_bound,
```

Comment:
I would move the `size_type` to the front of the tparam list so you don't have to specify the `cg_type` as it can be inferred from `g`.

Reply:
It's about the intention of the API and the syntax consistency.

Reply:
Sure, the `g` parameter should be the first one for consistency reasons but we can still use a different ordering for the tparam list, i.e., the one that lets us make use of automatic type inference. The only tparam that cannot be inferred is the result size type so specifying the CG type is redundant.

Reply:
It's an internal API so I don't want to bikeshed too much. I'm okay with merging it as is.